src/tools/wasm-merge.cpp

```diff
@@ -110,6 +110,27 @@ namespace {
 // have it as a global rather than pass it around all the time.
 Module merged;
 
+// Name conflicts on functions etc. are resolved by renaming things in a way
+// that only matters internally. Conflicting export names, however, are
+// observable, and so the user must decide how they want wasm-merge to handle
+// that.
+enum ExportMergeMode {
+  // Error on name conflicts. This is the least surprising mode, and the one
+  // used by default.
+  ErrorOnExportConflicts,
+  // Rename conflicting exports. Later exports will get a suffix added to them
+  // to make them unique. For example, this is useful if you merge several
+  // modules that each have a "main" export, and it's fine if those are renamed
+  // to "main", "main_1", "main_2" etc. - you'll decide when to call each of
+  // those and do so at the right time.
+  RenameExportConflicts,
+  // Keep the exports from the first module and remove all others. For example,
+  // this is useful when the first module is the main program and the others are
+  // libraries of code that it uses, but that do not have any exports intended
+  // to be used by anyone other than the main program.
+  KeepOnlyFirstModuleExports,
```

Comment:
Sorry I didn't catch this earlier when you were describing the 3 modes you had in mind!

Reply:
Interesting. That would allow more things to work. But it would also add more effort for the user in the common case this is meant to address, which is a main module + side modules, since we never want the side module's exports to remain. But maybe that's just more work we should expect the user to do, hmm... they also likely would want to prune some main module exports.